Functions/xASL_bids_MergeNifti_Merge.m

```diff
@@ -0,0 +1,148 @@
+            error('Dimensionality incorrect for this ASL NIfTI file');
+        end
+
+        % Compare size of the new file and if similar to the previous than concatenate, otherwise report an error
+        if iFile == 1
+            sizeFirst = size(tempIM);
+            sizeFirst = sizeFirst(1:3);
+        else
+            sizeNew = size(tempIM);
+            sizeNew = sizeNew(1:3);
+            if ~isequal(sizeNew, sizeFirst)
+                bStatus = 0;
+            end
+        end
+
+        if bAlternatingControlLabel
+            % Always interlace the two following files
+            % For the first file, create the interleaved first volume
+            if iFile == 1
+                lengthFirst = size(tempIM,4);
+                IM = zeros([sizeFirst,lengthFirst*2]);
+                IM(:,:,:,1:2:end) = tempIM;
+            elseif mod(iFile,2)
+                % For odd files, create a new interleaved addition
+                lengthFirst = size(tempIM,4);
+                IM(:,:,:,end+1:end+2*lengthFirst) = zeros([sizeFirst,lengthFirst*2]);
+                IM(:,:,:,end+2:2:end+2*lengthFirst) = tempIM;
+            else
+                % For even files - fill in the interleave spaces
+                IM(:,:,:,end-2*lengthFirst+2:2:end) = tempIM;
+            end
+        else
+            % Simply merge files in the order in which they come
+            if iFile==1
+                % Get the size of the first file
+                IM = tempIM;
+            else
+                if bStatus
+                    IM(:,:,:,end+1:end+size(tempIM,4)) = tempIM;
+                end
+            end
+        end
+
+        % Check for the path to JSON if existing and keep only the first existing JSON
+        if isempty(firstJSON)
+            [Fpath, Ffile] = xASL_fileparts(NiftiPaths{indexSortedFile(iFile)});
+            pathJSON = fullfile(Fpath,[Ffile '.json']);
+            if exist(pathJSON, 'file')
+                firstJSON = pathJSON;
+            end
+        end
+    end
+
+    %% If at the end and all went well
+    if bStatus
+        fprintf('Warning: concatenating multiple NIfTIs & jsons as output from dcm2niiX\n');
+        % Save the concatenated file to a given name
+        pathOut = fullfile(Fpath,[nameMerged '.nii']);
+        xASL_io_SaveNifti(NiftiPaths{indexSortedFile(1)}, pathOut, IM, [], 0);
+        % Special treatment for Hadamard encoded files
+        EchoTimes = cell(size(NiftiPaths,2),1);
+        for iFileCheck = 1:size(NiftiPaths,2)
+            % Get JSON
+            [jsonPathX, jsonNameX] = xASL_fileparts(NiftiPaths{iFileCheck});
+            if exist(fullfile(jsonPathX, [jsonNameX '.json']),'file')
+                tmpCheckJSON = spm_jsonread(fullfile(jsonPathX, [jsonNameX '.json']));
+                % Check EchoTimes
+                if isfield(tmpCheckJSON,'SeriesDescription')
+                    isHadamardFME = ~isempty(regexp(char(tmpCheckJSON.SeriesDescription),...
```

Comment:
@jan-petr 
This is only valid for Siemens, needs to be specified here otherwise it will generate errors for Hadamard of other vendors.
Also, what happens if `SeriesDescription` doesn't exist?